### PR TITLE
[runtimes] Ml 1373 - Validate that the service really exists and raise an error if not.

### PR DESCRIPTION
**Pull request by davesh0812**

### CODE REVIEW

File: mlrun/k8s_utils.py [hunk 1]
```diff
@@ -475,6 +475,37 @@ def get_project_secret_data(self, project, secret_keys=None, namespace=""):
...
+            if not states or service.status.phase in states
+        ]
+
+    def is_service_exist(self, name, namespace=None, is_release_name: bool = True):
+        items = self.list_services(namespace=namespace)
+        for item in items:
+            if (
+                is_release_name
+                and item.metadata.annotations is not None
+                and "meta.helm.sh/release-name" in item.metadata.annotations
```
Comment (guy1992l) [inline mlrun/k8s_utils.py:499]: please add some comments so one who is not familiar with how services are stored in Kubernetes will understand how you checked the name exists (what is release name, paste an example how it looks like in the yaml for example)

File: mlrun/k8s_utils.py [hunk 2]
```diff
@@ -475,6 +475,37 @@ def get_project_secret_data(self, project, secret_keys=None, namespace=""):
...
+            ):
+                if name == item.metadata.annotations["meta.helm.sh/release-name"]:
```
Comment (guy1992l) [inline mlrun/k8s_utils.py:501]: you can simply add it the the list of 'and's in the first if. And is checked from left to right and the first to return False will stop the check so if the `"meta.helm.sh/release-name"` key is not in the dictionary the `name == item.metadata.annotations["meta.helm.sh/release-name"]` won't be checked.

File: mlrun/k8s_utils.py (hunk 1 above)
Comment (guy1992l) [inline mlrun/k8s_utils.py:499]: Verify with @Hedingber its a valid / robust check. Maybe add a configuration from `mlconf` if its not something set in stone.

File: mlrun/runtimes/remotesparkjob.py [hunk 3]
```diff
@@ -129,6 +131,11 @@ def spec(self, spec):
 
     def with_spark_service(self, spark_service, provider=RemoteSparkProviders.iguazio):
         """Attach spark service to function"""
+        # check for existence of spark_services
+        if not get_k8s_helper().is_service_exist(name=spark_service):
+            raise MLRunInvalidArgumentError(
+                f"spark service named {spark_service} doesn't exist"
```
Comment (guy1992l) [inline mlrun/runtimes/remotesparkjob.py:137]: Be nicer in the error message :)

`The spark service provided '{}' does not exist. Please notice the name should be given as it is shown in the Iguazio UI.`

Something like that

**Review by guy1992l — CHANGES_REQUESTED** [review]
Small changes and its perfect. Ask Hedi to look into the service check and use `git checkout` on `mlrun/feature_store/steps.py` to remove it from this commit.

File: mlrun/runtimes/remotesparkjob.py [hunk 4]
```diff
@@ -129,6 +131,12 @@ def spec(self, spec):
 
     def with_spark_service(self, spark_service, provider=RemoteSparkProviders.iguazio):
         """Attach spark service to function"""
+        # check for existence of spark_services
```
Comment (yaronha) [inline mlrun/runtimes/remotesparkjob.py:134]: @guy1992l @davesh0812 the test should not be in the client (w/o k8s access), rather in the server side when this value is used for deployment (and there is k8s)

**Pull request closed without merging**